### ISSUE TITLE
End-of-options signal to `source` inadvertently triggers SC1091 

### ISSUE THREAD

**Issue by cueedee** [koalaman/shellcheck#1518]
#### For bugs
- Rule id: `SC1091`;
- My `shellcheck` version: `0.6.0`;
- [✓] [The rule's wiki page](https://shellcheck.net/wiki/SC1091) does not already cover this;
- [✓] I tried on shellcheck.net and verified that this is still a problem on the latest commit;

#### Here's a snippet or screenshot that shows the problem:

```bash
#!/usr/bin/bash

i="-c"             ## This file's name is indeed "-c";
source -- "${i}"   ## The `--` argument is needed to make `source` grok `-c` as a filename argument and not as an (invalid) option to `source`;
                   ## This is just a demo; in reality the `"${i}"` argument would be the loop variable feeded from a glob;
```

#### Here's what shellcheck currently says:

> ```
> source -- "${i}"
>        ^-- SC1091: Not following: -- was not specified as input (see > shellcheck -x).
> ```

#### Here's what I wanted or expected to see:

Nothing since `--` is not an input at all.

**Comment by koalaman** [comment]
It looks like ksh/bash/ash/dash all accept `--` even though they accept no options. This has now been fixed in a simplistic way by simply skipping a literal `--` if it exists. Thanks!

**Comment by cueedee** [comment]
Righ! As https://www.gnu.org/software/bash/manual/html_node/Shell-Builtin-Commands.html puts it:

> Unless otherwise noted, each builtin command documented as accepting options preceded by ‘`-`’ accepts ‘`--`’ to signify the end of the options. The `:`, `true`, `false`, and `test`/`[` builtins do not accept options and do not treat ‘`--`’ specially. The `exit`, `logout`, `return`, `break`, `continue`, `let`, and `shift` builtins accept and process arguments beginning with ‘`-`’ without requiring ‘`--`’. Other builtins that accept arguments but are not specified as accepting options interpret arguments beginning with ‘`-`’ as invalid options and require ‘`--`’ to prevent this interpretation.
